tianshou/policy/base.py

```diff
@@ -231,7 +231,8 @@ def post_process_fn(self, batch: Batch,
         usage is to update the sampling weight in prioritized experience
         replay. Check out :ref:`policy_concept` for more information.
         """
-        if isinstance(buffer, PrioritizedReplayBuffer):
+        if isinstance(buffer, PrioritizedReplayBuffer) \
+                and hasattr(batch, 'weight'):
```

Comment:
That's strange. Does any PrioritizedReplayBuffer have no ``weight ``?

Reply:
Good point. I would add that as a requirement.